language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureHelpUtil.java

```diff
@@ -497,28 +498,38 @@ private static boolean withinTextRange(int position, @Nonnull NonTerminalNode no
-            String funcName;
-            Predicate<Symbol> symbolPredicate =
-                    symbol -> symbol.kind() == FUNCTION || symbol.kind() == VARIABLE || symbol.kind() == PARAMETER;
-            List<Symbol> filteredContent;
-            if (nameReferenceNode.kind() == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
-                QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) nameReferenceNode;
-                funcName = (qNameRef).identifier().text();
-                filteredContent = QNameReferenceUtil.getModuleContent(context, qNameRef,
-                                                                      symbolPredicate
-                                                                              .and(symbol -> symbol.getName().orElse("")
-                                                                                      .equals(funcName)));
-            } else {
-                funcName = ((SimpleNameReferenceNode) nameReferenceNode).name().text();
-                List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition());
-                filteredContent = visibleSymbols.stream()
-                        .filter(symbolPredicate.and(symbol -> symbol.getName().get().equals(funcName)))
-                        .collect(Collectors.toList());
-            }
+            return getFunctionSymbol(nodeAtCursor, context);
+        }
+        return getFunctionSymbol(nodeAtCursor, context);
 
-            return filteredContent.stream().findAny();
+    }
+
+    private static Optional<? extends Symbol> getFunctionSymbol(FunctionCallExpressionNode node,
+                                                                SignatureContext context) {
+        NameReferenceNode nameReferenceNode = node.functionName();
+        String funcName;
+        Predicate<Symbol> symbolPredicate =
+                symbol -> symbol.kind() == FUNCTION || symbol.kind() == VARIABLE || symbol.kind() == PARAMETER;
+        List<Symbol> filteredContent;
+        if (nameReferenceNode.kind() == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+            QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) nameReferenceNode;
+            funcName = (qNameRef).identifier().text();
+            filteredContent = QNameReferenceUtil.getModuleContent(context, qNameRef,
+                    symbolPredicate
+                            .and(symbol -> symbol.getName().orElse("")
+                                    .equals(funcName)));
+        } else {
+            funcName = ((SimpleNameReferenceNode) nameReferenceNode).name().text();
+            List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition());
+            filteredContent = visibleSymbols.stream()
+                    .filter(symbolPredicate.and(symbol -> symbol.getName().isPresent() && symbol.getName().get().equals(funcName)))
```

Comment:
Checkstyle failure due to number of characters on this line

Reply:
Done